gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,436 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+import matplotlib.pylab as plt
+from scipy import stats
+
+__all__ = ['find_confidence_interval_gauss',
+           'find_confidence_interval_poisson'
+           'construct_confidence_belt_PDFs',
+           'get_upper_and_lower_limit',
+           'fix_upper_and_lower_limit',
+           'find_limit',
+           'find_average_upper_limit',
+           'construct_confidence_belt']
+
+
+def find_confidence_interval_gauss(mu, sigma, x_bins, fCL):
+    """Analytically find confidence interval for Gaussian with boundary 
+       at the origin
+
+    Parameters
+    ----------
+    mu : double
+        Mean of the Gaussian
+    sigma : double
+        Width of the Gaussian
+    x_bins : array-like
+        Bins in x
+    fCL : double
+        Desired confidence level
+
+    Returns
+    -------
+    : double
+        Beginning of the confidence interval
+    : double
+        End of the confidence interval
+    """
+
+    dist = stats.norm(loc=mu, scale=sigma)
+
+    x_bin_width = x_bins[1] - x_bins[0]
+
+    p = []
+    r = []
+
+    for x in x_bins:
+        p.append(dist.pdf(x)*x_bin_width)
+        # This is the formula from the FC paper
+        if mu == 0 and sigma == 1:
+            if x < 0:
+                r.append(numpy.exp(x*mu-mu*mu*0.5))
+            else:
+                r.append(numpy.exp(-0.5*numpy.power((x-mu),2)))
+        # This is the more general formula
+        else:
+            # Implementing the boundary condition at zero
+            muBest     = max(0, x)
+            probMuBest = stats.norm.pdf(x, loc=muBest, scale=sigma)
+            if probMuBest == 0.0:
+              r.append(0.0);
+            else:
+              r.append(p[-1]/probMuBest)
+
+    p = numpy.asarray(p)
+    r = numpy.asarray(r)
+
+    if sum(p) < fCL:
+        print("Bad choice of x-range for this mu!")
+        print("Not enough probability in x bins to reach confidence level!")
+
+    rank = stats.rankdata(-r, method='dense')
+
+    index_array = numpy.arange(x_bins.size)
+
+    rank_sorted, index_array_sorted = zip(*sorted(zip(rank, index_array)))
+
+    index_min = index_array_sorted[0]
+    index_max = index_array_sorted[0]
+
+    p_sum = 0
+
+    for i in xrange(len(rank_sorted)):
```

Comment:
xrange doesn't exist in Python 3, I think this will fail to import.

My recommendation would be to use range, which is like xrange on Python 3, and on Python 2 makes lists.

If memory / speed efficiency really is a concern, you can put this at the top:
```
from astropy.extern.six.moves import range
```
Then it will be the efficient variant on both Python 2 and 3.